### PR TITLE
Settings authorization feature

### PR DESCRIPTION
**Pull request by thomasmaas**
fixes tests and code for member users that only have access to settings (which is kind of crazy if you think of it but that's another story)

### CODE REVIEW

File: app/views/provider/admin/dashboards/_developers_navigation.html.slim [hunk 1]
```diff
@@ -17,6 +17,7 @@ nav.DashboardNavigation
                                                   admin_buyers_accounts_path(search: {state: 'pending'}),
                                                   'Pending'
 
+    - elsif can?(:manage, :settings)
```
Comment (Martouta) [inline app/views/provider/admin/dashboards/_developers_navigation.html.slim]: What is this doing? There is nothing inside it besides a comment, right?

File: features/support/paths.rb [hunk 2]
```diff
@@ -575,6 +587,9 @@ def path_to(page_name, *args)
     when "finance settings"
       admin_finance_settings_path
 
+    when "the finance settings"
+      admin_finance_settings_path
```
Comment (Martouta) [inline features/support/paths.rb]: I don't think we need this twice 🤔 It is the same as the previous one. So maybe we should remove the previous one and rename them all to "the finance settings"

File: features/support/paths.rb [hunk 3]
```diff
@@ -467,6 +467,18 @@ def path_to(page_name, *args)
...
       admin_site_dns_path
 
+    when 'the spam protection page'
+      edit_admin_site_spam_protection_path
+
+    when 'the xss protection page'
+      edit_admin_site_developer_portal_path
+
+    when 'the sso integrations page'
+      provider_admin_authentication_providers_path
```
Comment (Martouta) [inline features/support/paths.rb]: Kinda the same as https://github.com/3scale/porta/pull/175/files#r226311032
This already exists in:
https://github.com/3scale/porta/blob/23b4398ad7a97fb4b0a66f112deca50c7280203b/features/support/paths.rb#L25-L26

File: features/support/paths.rb (hunk 2 above)
Comment (thomasmaas) [reply features/support/paths.rb]: that will break other tests so I'm gonna leave it.

Comment (thomasmaas) [reply features/support/paths.rb]: well we should make the 'the' optional

Comment (Martouta) [reply features/support/paths.rb]: @thomasmaas It shouldn't break anything, it should be a very straightforward change 🤔 Do you want me to do it quickly and check it doesn't break anything (through Jenkins/CircleCI) ?

**Pull request merged**